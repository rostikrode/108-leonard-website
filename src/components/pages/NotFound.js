import React from 'react';
import '../../styles/NotFound.css';

const NotFound = () => {
  window.gtag('config', 'UA-113369414-1', {
    'page_title': 'Page Not Found',
    'page_location': window.location.href,
    'page_path': window.location.pathname
  });

  /** meta data for page */
  document.title = '404 Page Not Found';
  if(document.getElementsByTagName('meta').description) {
    document.getElementsByTagName('meta').description.content = 'Your page was not found';
    document.querySelector("meta[property='og:description']").content = 'Your page was not found';
    document.querySelector("meta[property='og:title']").content = '404 Page Not Found';
  }
  if (document.querySelector("link[rel='canonical']")) {
    document.querySelector("link[rel='canonical']").href = 'https://108leonard.com/404'
    document.querySelector("meta[property='og:url']").content = 'https://108leonard.com/404'
  }
  var viewport = document.querySelector("meta[name=viewport]");
  if(viewport) {
    viewport.setAttribute('content', 'width=device-width, initial-scale=1, user-scalable=1');
  }

  return (
    <div className="mid-block">
      <h1>
        404 page not found :(
      </h1>
    </div>
  );
}

export default NotFound;