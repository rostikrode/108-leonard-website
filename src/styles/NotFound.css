.mid-block {
  width: 100%;
  height: 100%;
  padding-top: 5em;
  display: flex;
  justify-content: center;
  align-items: center;
}