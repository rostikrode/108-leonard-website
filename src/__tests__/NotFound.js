import 'babel-polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NotFound from '../components/pages/NotFound';
import { MemoryRouter } from 'react-router-dom';


it('renders without crashing', () => {
  const tree = renderer.create(
    <MemoryRouter><NotFound /></MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});