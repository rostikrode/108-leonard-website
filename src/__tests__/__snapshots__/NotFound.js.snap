// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="mid-block"
>
  <h1>
    404 page not found :(
  </h1>
</div>
`;
